activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,23 @@
+*   `ActiveRecord::Encryption::Encryptor` now supports a `:compressor` option to customize the compression algorithm used.
+
+    ```ruby
+    module ZstdCompressor
+      def self.deflate(data)
+        Zstd.compress(data)
+      end
+
+      def self.inflate(data)
+        Zstd.decompress(data)
+      end
+    end
+
+    class User
+      encrypts :name, encryptor: ActiveRecord::Encryption::Encryptor.new(compressor: ZstdCompressor)
```

Comment:
This API doesn't feel right. Having to initialize a new Encryptor in your app to set the compressor feels cumbersome and un-Rails like. I have two suggestions:
 
1) I think if we want a compressor option for every model then Rails should do the initialization for the app. The AR code would check if the compressor option is set and then Rails would do the initialization for the application. Then it's a much cleaner API.

```ruby
encrypts :name, compressor: ZstdCompressor
```

2) Rails should provide a config option so that applications can set all encryptors to use the same custom compressor.

Reply:
@eileencodes Thank you for review! Your talk at the 2023 Rails conference inspired me to contribute to Rails, and I'm glad I got a review 😄 

1. This API seems nicer than the earlier one. I'll implement it. Additionally, I think we should accept the compress option in the encrypt method, as introduced in https://github.com/rails/rails/pull/50876. I'll handle that in this PR.
2. I've already implemented this feature in Config. Should I just add it to the CHANGELOG?

Reply:
@heka1024 New config parameters require a little more documentation work than other features. Specifically, you'll need to update `guides/source/configuring.md`.

There is a linting tool you should run to make sure `configuring.md` changes are correct, run `tools/railspect configuration .`